lib/mercury/cps.rb

```diff
@@ -35,18 +35,18 @@ def run(*args, &k)
     #   Cps and returns a Cps
     def and_then(&pm)
       Cps.new do |*args, &k|
-        self.run(*args) do |*args2|
-          next_cps = pm.call(*args2)
-          next_cps.is_a?(Cps) or raise "'and_then' block did not return a Cps object. Did you want 'and_lift'? at #{pm.source_location}"
+        run(*args) do |*args2|
+          next_cps = yield(*args2)
+          next_cps.is_a?(Cps) || raise("'and_then' block did not return a Cps object. Did you want 'and_lift'? at #{pm.source_location}")
```

Comment:
Using parentheses for `raise` is in violation of [this rule](https://github.com/bbatsov/ruby-style-guide#no-dsl-parens) from the style guide, as well as contradicts every example usage of `raise` in the guide.